Result:

```
AssertionError: 13 != 2 : [b"<string>:3: EncodingWarning: 'encoding' argument not specified.", b'Traceback (most recent call last):', b'  File "<string>", line 3, in <module>', b'  File "C:\\Workspace\\buildarea\\pull_request.linaro-win-arm64.nondebug\\build\\Lib\\subprocess.py", line 533, in run', b'    with Popen(*popenargs, **kwargs) as process:', b'         ^^^^^^^^^^^^^^^^^^^^^^^^^^^', b'  File "C:\\Workspace\\buildarea\\pull_request.linaro-win-arm64.nondebug\\build\\Lib\\subprocess.py", line 998, in __init__', b'    self._execute_child(args, executable, preexec_fn, close_fds,', b'    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^', b'  File "C:\\Workspace\\buildarea\\pull_request.linaro-win-arm64.nondebug\\build\\Lib\\subprocess.py", line 1467, in _execute_child', b'    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,', b'                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^', b'FileNotFoundError: [WinError 2] The system cannot find the file specified']
```
On multiple lines:
```
<string>:3: EncodingWarning: 'encoding' argument not specified.
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "C:\Workspace\buildarea\pull_request.linaro-win-arm64.nondebug\build\Lib\subprocess.py", line 533, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Workspace\buildarea\pull_request.linaro-win-arm64.nondebug\build\Lib\subprocess.py", line 998, in __init__       
    self._execute_child(args, executable, preexec_fn, close_fds,
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Workspace\buildarea\pull_request.linaro-win-arm64.nondebug\build\Lib\subprocess.py", line 1467, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

@methane

--- a/Lib/test/test_subprocess.py
+++ b/Lib/test/test_subprocess.py
@@ -1736,16 +1736,15 @@ class RunFuncTestCase(BaseTestCase):
     def test_encoding_warning(self):
         code = textwrap.dedent("""\
             from subprocess import *
-            args = ["echo", "hello"]
-            run(args, text=True)
-            check_output(args, text=True)
+            run("echo hello", shell=True, text=True)
+            check_output("echo hello", shell=True, text=True)
             """)
         cp = subprocess.run([sys.executable, "-Xwarn_default_encoding", "-c", code],
                             capture_output=True)
         lines = cp.stderr.splitlines()
         self.assertEqual(len(lines), 2, lines)
-        self.assertTrue(lines[0].startswith(b"<string>:3: EncodingWarning: "))
-        self.assertTrue(lines[1].startswith(b"<string>:4: EncodingWarning: "))
+        self.assertTrue(lines[0].startswith(b"<string>:2: EncodingWarning: "))
+        self.assertTrue(lines[1].startswith(b"<string>:3: EncodingWarning: "))
 
 
 def _get_test_grp_name():

